### PR TITLE
Fixing Linters and tests in file-stores (ftp, sftp, s3)

### PR DESCRIPTION
**Pull request by coolwednesday**
-   prequisite to closing #964
-   Some methods previously included required a running container for validation. These methods have been temporarily removed and, if needed, their mocking will be addressed separately in a future PRs.

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

**Review by Umang01-hash — DISMISSED** [review]

File: pkg/gofr/datasource/file/ftp/fs.go [hunk 1]
```diff
@@ -14,7 +14,11 @@ import (
 	file_interface "gofr.dev/pkg/gofr/datasource/file"
 )
 
-const appFtpStats = "app_ftp_stats"
+const (
+	appFtpStats   = "app_ftp_stats"
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/fs.go:18]:
```suggestion
	appFTPStats   = "app_ftp_stats"
```

It's an acronym, fixing an unexported constant won't cause problem

File: pkg/gofr/datasource/file/ftp/fs_test.go [hunk 2]
```diff
@@ -10,11 +10,18 @@ import (
...
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/jlaffaye/ftp"
 	"github.com/stretchr/testify/assert"
 	"go.uber.org/mock/gomock"
 )
 
+var (
+	errMockErr  = errors.New("mocked error")
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/fs_test.go:21]:
```suggestion
	errMockSentinel  = errors.New("mocked error")
```

Or anything but "error mock error"

File: pkg/gofr/datasource/file/s3/file.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ import (
 )
 
 type s3file struct {
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/file.go:20]:
```suggestion
type s3File struct {
```

File: pkg/gofr/datasource/file/s3/file_parse.go [hunk 4]
```diff
@@ -141,3 +144,89 @@ func (f *textReader) Scan(i interface{}) error {
...
+}
+
+// IsDir checks if the FileInfo describes a directory.
+//
+// This method returns true if the FileInfo object represents a directory; otherwise, it returns false.
+// It is specifically used to determine the type of the file system object represented by the FileInfo.
+//
+// Note:
+//   - This method should be called on a FileInfo instance obtained from a Stat or ReadDir operation.
+//   - The FileInfo interface is used to describe file system objects, and IsDir is one of its methods
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/file_parse.go:221]:
```suggestion
//   - The [FileInfo] interface is used to describe file system objects, and IsDir is one of its methods
```

File: pkg/gofr/datasource/file/s3/fs.go [hunk 5]
```diff
@@ -20,9 +18,23 @@ import (
...
+
+// Conn struct embeds the *ftp.ServerConn returned by ftp server on successful connection.
+type conn struct {
+	*s3.Client
+}
+
 type fileSystem struct {
 	s3file
-	conn    *s3.Client
+	conn    s3conn
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/fs.go:37]: You define conn, then use s3conn

Maybe you could rename conn to s3Client or s3ClientConn

File: pkg/gofr/datasource/file/s3/fs_dir.go [hunk 6]
```diff
@@ -289,6 +303,90 @@ func (f *fileSystem) renameDirectory(st, msg *string, oldPath, newPath string) e
...
+	filetype := TypeFile
+
+	// Here we assume the user passes "0filePath" in case it wants to get fileinfo about a binary file instead of a directory
+	if path.Ext(name) == "" {
+		filetype = TypeDirectory
+
+		if name[0] == '0' {
+			name = name[1:]
+			filetype = TypeFile
+		}
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/fs_dir.go:334]: strings.CutPrefix could help

File: pkg/gofr/datasource/file/s3/fs_dir.go [hunk 7]
```diff
@@ -289,6 +303,90 @@ func (f *fileSystem) renameDirectory(st, msg *string, oldPath, newPath string) e
...
+		Message:   &msg,
+	}, time.Now())
+
+	filetype := TypeFile
+
+	// Here we assume the user passes "0filePath" in case it wants to get fileinfo about a binary file instead of a directory
+	if path.Ext(name) == "" {
+		filetype = TypeDirectory
+
+		if name[0] == '0' {
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/fs_dir.go:331]: A comment would be appreciated

File: pkg/gofr/datasource/file/s3/fs_dir.go [hunk 8]
```diff
@@ -289,6 +303,90 @@ func (f *fileSystem) renameDirectory(st, msg *string, oldPath, newPath string) e
...
+		}
+
+		return nil, nil
+	}
+
+	return &s3file{
+		conn:         f.conn,
+		logger:       f.logger,
+		metrics:      f.metrics,
+		size:         *res.Contents[0].Size,
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/fs_dir.go:383]: These access to *res.Contents[0] everywhere in this method, causes me to wonder about panic.

Maybe you could store into a variable, then use only the variable

File: pkg/gofr/datasource/file/s3/interface.go [hunk 9]
```diff
@@ -12,6 +14,15 @@ type Logger interface {
 	Errorf(pattern string, args ...interface{})
 }
 
+type s3conn interface {
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/s3/interface.go:17]: Can you rename it? Thanks


```suggestion
type s3Conn interface {
```

File: pkg/gofr/datasource/file/s3/fs.go (hunk 5 above)
Comment (ccoVeille) [reply pkg/gofr/datasource/file/s3/fs.go:37]: Yes, anything that not increase the confusion like having s3conn and conn being different things

File: pkg/gofr/datasource/file/ftp/fs.go [hunk 10]
```diff
@@ -80,7 +84,7 @@ func (f *fileSystem) Connect() {
 	var status string
 
 	ftpBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
-	f.metrics.NewHistogram(appFtpStats, "Response time of File System operations in milliseconds.", ftpBuckets...)
+	f.metrics.NewHistogram(appFTPStats, "Response time of File System operations in milliseconds.", ftpBuckets...)
```
Comment (aryanmehrotra) [inline pkg/gofr/datasource/file/ftp/fs.go:87]: time should be in microseconds as we are exporting time in microseconds

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]
Lets create 3 different PRs.

File: pkg/gofr/datasource/file/s3/fs.go [hunk 11]
```diff
@@ -20,9 +18,23 @@ import (
 	file "gofr.dev/pkg/gofr/datasource/file"
 )
 
+const (
+	TypeFile      = "file"
```
Comment (aryanmehrotra) [inline pkg/gofr/datasource/file/s3/fs.go:22]: why exported?

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]
Lets create 3 different PRs.

**Comment by aryanmehrotra** [comment]
Closing as separated PRs are being created.

**Comment by ccoVeille** [comment]
Split to

- #1265
-

**Pull request closed without merging**